### PR TITLE
Add timeout and http method for fetch.

### PR DESCRIPTION
**Pull request by raych1**
Use head to fetch;
Set timeout and change retry;
Upgrade other dependencies.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -74,27 +74,27 @@ export const pathJoin = (dir: string, value: string): string => {
...
           return true
-        },
-        {
-          retries: 3,
-          factor: 1
+        } 
+        if (retries === retryTimes) {
+          break
+        }
+      } catch (e) {
```
Comment (lirenhe) [inline src/index.ts:99]: I don't think we need to retry on all errors.

File: src/index.ts [hunk 2]
```diff
@@ -74,27 +74,27 @@ export const pathJoin = (dir: string, value: string): string => {
...
-
+    let retries = 0
+    const retryTimes = 3
+    while (retries++ < retryTimes) {
+      try {
+        const { status } = await fetch(dir, {
+          method: "HEAD",
+          timeout: 60 * 1000
+        })
+        if (status === 200) {
```
Comment (lirenhe) [inline src/index.ts:93]: use 2xx instead of 200?

File: package.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "main": "dist/index.js",
   "scripts": {
     "tsc": "tsc",
-    "test": "tsc && nyc mocha ./dist/test/test.js -t 10000",
+    "test": "tsc && nyc mocha ./dist/test/test.js -t 300000",
     "prepack": "npm install && tsc"
   },
   "nyc": {
```
Comment (ruowan) [inline package.json:11]: bump version and changelog?

File: src/index.ts [hunk 4]
```diff
@@ -74,27 +74,27 @@ export const pathJoin = (dir: string, value: string): string => {
...
           return true
-        },
-        {
-          retries: 3,
-          factor: 1
+        }
```
Comment (ruowan) [inline src/index.ts]: For not 200 status trials, can we collect the result and report it after retry finished.

**Review by ruowan — APPROVED** [review]

**Review by lirenhe — APPROVED** [review]

**Pull request merged**